api/v1alpha1/healthcheck_types.go

```diff
@@ -0,0 +1,149 @@
+// Copyright Envoy Gateway Authors
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// HealthCheck defines the health check configuration.
+type HealthCheck struct {
```

Comment:
are all these values `required` ?
imo they should all be optional with sane defaults except for `healthChecker`

Reply:
Fixed.